File: 백준/Silver/1015. 수열 정렬/수열 정렬.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const N = Number(input[0])
const A = input[1].split(" ").map(Number)

const B = [...A].sort((a, b) => a-b)
const P = Array(N).fill(-1)
A.forEach((a, index) => {
    const targetIndex = B.indexOf(a)
    if (!P.includes(targetIndex)) {
      P[index] = targetIndex
    } else {
        let startIndex = targetIndex+1
       while(true){
           if(B[startIndex] === a && !P.includes(startIndex)){
               P[index] = startIndex
               break
           }
           startIndex++
           if(startIndex===N){
               break
           }
       }
    }
  
})

console.log(P.join(' '))

File: 백준/Silver/1015. 수열 정렬/수열 정렬.py
import sys
input = sys.stdin.readline

# 입력 처리
N = int(input().strip())
A = list(map(int, input().strip().split()))

# B = [1,2,3]이 되어야 함
B = sorted(A)
visited = [False] * N
P = []

for value in A:
    for i in range(N):
        if visited[i]:
            continue
        if B[i] == value:
            P.append(i)
            visited[i] = True
            break

print(" ".join(map(str, P)))
